core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -202,9 +208,9 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
             }
             otherDocCount += terms.getSumOfOtherDocCounts();
             final long thisAggDocCountError;
-            if (terms.getBucketsInternal().size() < getShardSize() || InternalOrder.isTermOrder(order)) {
+            if (terms.getBucketsInternal().size() < getShardSize() || Terms.Order.isTermOrder(order)) {
                 thisAggDocCountError = 0;
-            } else if (InternalOrder.isCountDesc(this.order)) {
+            } else if (Terms.Order.isCountDesc(this.order)) {
```

Comment:
The `_count` order is specific to the `terms` agg (code treats it like the `_term` order). Also this method compares it directly to `Terms.Order#COUNT_DESC` instance so thats why this code references `Terms.Order` instead of something in `BucketOrder`.

It might be possible to move the `_count` order code into a common place like `BucketOrder` but that would require more changes (looks like `terms` and `histogram` aggs use a different `byte id` for `_count` order serialization.)